src/utils/appData.ts

```diff
@@ -21,12 +24,52 @@ async function getValidator(): Promise<{ ajv: Ajv; validate: ValidateFunction }>
   return { ajv, validate }
 }
 
-export async function validateAppDataDocument(appDataDocument: any): Promise<ValidationResult> {
+async function getSerializedCID(hash: string): Promise<void | string> {
+  const cidVersion = 0x1 //cidv1
+  const codec = 0x70 //dag-pb
+  const type = 0x12 //sha2-256
+  const length = 32 //256 bits
+  const _hash = hash.replace(/(^0x)/, '')
+
+  const hexHash = fromHexString(_hash)
+
+  if (!hexHash) return
+
+  const uint8array = Uint8Array.from([cidVersion, codec, type, length, ...hexHash])
+  const { CID } = await import('multiformats/cid')
+  return CID.decode(uint8array).toV0().toString()
+}
+
+async function loadIpfsFromCid(cid: string): Promise<AppDataDoc> {
+  const { default: fetch } = await import('cross-fetch')
+  const response = await fetch(`${DEFAULT_IPFS_URI}/${cid}`)
```

Comment:
i would expect that if `DEFAULT_IPFS_URI` is the default, is because there's a way to override. Maybe it can be part of the init params of the Sdk(`CowContext`)?